network/default.go

```diff
@@ -681,14 +706,20 @@ func (n *network) processCtrlChan(listener tunnel.Listener) {
 						Metric:  int(event.Route.Metric),
 					}
 					// set the route metric
-					n.node.RLock()
-					n.setRouteMetric(&route)
-					n.node.RUnlock()
-					// throw away metric bigger than 1000
-					if route.Metric > 1000 {
-						log.Debugf("Network route metric %d dropping node: %s", route.Metric, route.Router)
-						continue
+					n.RLock()
+					if link, ok := n.peerLinks[event.Route.Gateway]; ok {
+						// NOTE: should we change router.Route.Metric to int64?
+						if int(link.Length()) < route.Metric {
+							route.Metric = int(link.Length())
+						}
 					}
+					n.RUnlock()
+					// TODO: Are we dropping any routes?
```

Comment:
We need to come up with some way of dropping some routes. Otherwise we might end up with more routes than before.

Reply:
We're discarding routes if they're not in the peer graph right? So basically we might just want to recheck that it's not above the hop count of 3/4. 

Otherwise a route is probably not useful if the roundtrip is more than 5 seconds. So we may want to set the upper limit of the metric to that.

Reply:
We have agreed to just leave this for now. Since we discard routes beyond the peer graph at maxdepth 3 that creates a level of pruning. Once we have a better understanding of our new metric value then we'll decide how to proceed. This may require some moving average e.g if all things slow down the boundary is always moving higher.

We must also keep in mind Starlink!